resources/prosody-plugins/mod_muc_password_check.lua

```diff
@@ -0,0 +1,218 @@
+local jid = require "util.jid";
+local json = require "util.json";
+local async_handler_wrapper = module:require "util".async_handler_wrapper;
+local room_jid_match_rewrite = module:require "util".room_jid_match_rewrite;
+local starts_with = module:require "util".starts_with;
+local token_util = module:require "token/util".new(module);
+local inspect = require "inspect";
+local parse = require "net.url".parseQuery;
+
+
+-- option to enable/disable room API token verifications
+local enableTokenVerification
+= module:get_option_boolean("enable_password_token_verification", true);
+
+local muc_domain_base = module:get_option_string("muc_mapper_domain_base");
+if not muc_domain_base then
+    module:log("warn", "No 'muc_domain_base' option set, disabling password check endpoint.");
+    return ;
+end
+local muc_domain_prefix = module:get_option_string("muc_mapper_domain_prefix", "conference");
+
+local json_content_type = "application/json";
+
+--- Verifies room name, domain name with the values in the token
+-- @param token the token we received
+-- @param room_address the full room address jid
+-- @return true if values are ok or false otherwise
+function verify_token(token, room_address)
+    if not enableTokenVerification then
+        return true;
+    end
+
+    -- if enableTokenVerification is enabled and we do not have token
+    -- stop here, cause the main virtual host can have guest access enabled
+    -- (allowEmptyToken = true) and we will allow access to rooms info without
+    -- a token
+    if token == nil then
+        module:log("warn", "no token provided");
+        return false;
+    end
+
+    local session = {};
+    session.auth_token = token;
+    local verified, reason, msg = token_util:process_and_verify_token(session);
+    if not verified then
+        module:log("warn", "not a valid token %s %s", tostring(reason), tostring(msg));
+        return false;
+    end
+
+    -- accepting server tokens, which currently do not specify room permissions, so skip this check
+    -- if not token_util:verify_room(session, room_address) then
+    --     log("warn", "Token %s not allowed to join: %s",
+    --         tostring(token), tostring(room_address));
+    --     return false;
+    -- end
+
+    return true;
+end
+
+function handle_validate_room_password (event)
+    local request = event.request;
+    module:log("info","Request for room password validate: reqid %s", request.headers["request_id"])
+
+    if request.headers.content_type ~= json_content_type
+            or (not request.body or #request.body == 0) then
+        return 400;
+    end
+
+    local params = json.decode(event.request.body);
+    if not params then
+        return 400;
+    end
+
+    local conference = params["conferenceFullName"];
+    local passcode = params["passcode"];
+
+    if (not conference or not passcode) then
+        return { status_code = 400 };
```

Comment:
why do we return a table here vs just the code in the previous cases?

Reply:
No idea ... will change it. Most of the code was borrowed from another internal module.

Reply:
Actually, the status_code is expected by the handler. 
https://github.com/jitsi/jitsi-meet/blob/ffc412c18d7a6b4dc92e03d8a8a379031d24f4fc/resources/prosody-plugins/util.lib.lua#L149